DAQ_System/Source/main.cpp

```diff
@@ -1,10 +1,82 @@
 #include "mbed.h"
+#include "LSM303DLHC.h"
+#include <iostream>
+
+static float _lsm303Accel_MG_LSB     = 0.001F;
+const int sampleCount = 100;
+//this should be based on a calibrated value for 1.0 G
+const double gravity = 1.0;
+
+//the following can be used for calibration
+int loopNum = 0;
+double average = 0;
+double previousSamples[sampleCount];
+
+//can be used for data calibration 
+double averageData(double array[],int arraylength){
+	double total = 0,average = 0;
+	for(int k = 0;k<arraylength;k++){
+		total = total + array[k];
+	}
+	average = total/(arraylength);
+	return average;
+}
+//converts from raw data
+double accelerometerConvert(int input){
+    double output = (double)input * (double)_lsm303Accel_MG_LSB * gravity;
+    return output;
```

Comment:
- [ ] Instead of casting to a double, simply make `_lsm303Accel_MG_LSB` a double
- [ ] Rename `input`. Perhaps `raw_data` or `nominal_data`
- [ ] I think simply returning the expression will suffice, instead of allocating the `output` variable